limDaSeul/app.js

```diff
@@ -49,6 +49,124 @@ app.post("/users", async (req, res, next) => {
   res.status(201).json({message : "userCreated"});
 })
 
+//Create post table
+app.post("/posts", async(req, res, next) => {
+  const { title, content, posting_image, user_id } = req.body
+
+  await appDataSource.query(
+    `INSERT INTO posts(
+      title,
+      content,
+      posting_image,
+      user_id
+    ) VALUES(?, ?, ?, ?);
+    `,
+    [title, content, posting_image, user_id]
+  );
+  res.status(200).json({message : "postCreated"});
+})
+
+//Get post list
+app.get("/posts", async(req, res) => {
+
+  await appDataSource.manager.query(
+    `SELECT
+        u.profile_image as userProfileImage,
+        u.id as userID,
+        p.id as postingId,
+        p.posting_image as postingImageUrl,
+        p.content as postingContent
+      FROM users u, posts p
+      WHERE u.id = p.id
+      `,
+      (err, rows) =>  {
+         res.status(200).json(rows);
+    })
+})
+
+//Update post
+app.patch('/posts/update_des/:postID', async(req, res) => {
+  const postID = req.params.postID;
+  const {title, content, posting_image} = req.body
+
+  await appDataSource.query(
+    `UPDATE posts
+    SET
+      title = ?,
+      content = ?,
+      posting_image = ?
+    WHERE id = ${postID}
+   `,
+    [title, content, posting_image]
+    );
+
+  const posting = await appDataSource.query(
+    `SELECT
+        u.id as userID,
+        u.name as userName,
+        p.id as postingId,
+        p.title as postingTitle,
+        p.content as postingContent
+      FROM posts p
+      INNER JOIN users u
+      ON u.id = p.user_id
+      WHERE p.id
+    `);
+  res.status(201).json({data : posting[0]});
+});
```

Comment:
주석을 통해 해당 코드가 어떤 코드인지 구분하고 설명하는 것은 좋습니다!
다만 코드가 수정되었을 때 주석까지 신경써서 수정해주어야 하는 불편함이 있습니다.

코드가 수정되어도 주석이 수정되지 않았다면 다른 개발자에게 오히려 혼란을 야기할 수 있습니다.

따라서 위코드에서는 주석을 삭제하도록 가이드하고 있으니 다른 부분에 대한 주석도 삭제해주세요!

추가로 RESTful api는 엔드포인트에 동사형이 아닌 명사형을 작성하도록 권장하고 있습니다.

이미 HTTP method인 patch에는 특정 리소스를 수정하겠다는 의미가 내포되어 있으니 엔드포인트를 아래와 같이 수정해주세요!

변수명은 camelCase로 작성해주는 것을 신경써주세요!

```suggestion
//Update post
app.patch('/posts/:postId', async(req, res) => {
  const postId = req.params.postId;
  const {title, content, postingImage} = req.body

  await appDataSource.query(
    `UPDATE posts
    SET
      title = ?,
      content = ?,
      posting_image = ?
    WHERE id = ${postId}
   `,
    [title, content, postingImage]
    );

  const posting = await appDataSource.query(
    `SELECT
        u.id as userID,
        u.name as userName,
        p.id as postingId,
        p.title as postingTitle,
        p.content as postingContent
      FROM posts p
      INNER JOIN users u
      ON u.id = p.user_id
      WHERE p.id
    `);
  res.status(201).json({data : posting[0]});
});
```